tools/matchms/matchms_metadata_match.xml

```diff
@@ -0,0 +1,120 @@
+                <option value="difference" selected="true">difference</option>
+                <option value="equal_match">equal match</option>
+            </param>
+            <when value="difference">        
+                <param label="Metadata Match" name="metadata_match_field" type="select" display="radio"
+                    help="Metadata field to use for matching">
+                    <option value="retention_index" selected="true">Retention Index</option>
+                    <option value="retention_time">Retention Time</option>
+                    <option value="precursor_mz">Precursor Mz</option>
+                </param>
+                <param label="Tolerance" name="tolerance" type="float" value="0"
+                    help="Tolerance for matching the selected metadata field - only pairs within the tolerance will be contained in the output." />
+            </when>
+            <when value="equal_match">
+                <param label="Metadata Match" name="metadata_match_field" type="select" display="radio"
+                    help="Metadata field to use for matching">
+                    <option value="smiles" selected="true">SMILES</option>
+                    <option value="inchi">InChI</option>
+                    <option value="inchi_key">InChIKey</option>
+                </param>
+            </when>
+        </conditional>
+        <param label="join type" name="join_type" type="select" display="radio"
+            help="join type to use for appending scores">
+            <option value="left" selected="true">left</option>
+            <option value="inner">inner</option>
+        </param>
+    </inputs>
+
+    <outputs>
+        <data label="Match $matching.metadata_match_field results of ${on_string}" name="scores_out" format="json"/>
+    </outputs>
+
+    <tests>
+        <test>
+            <param name="queries" value="similarity/rcx_gc-ei_ms_subset.msp" ftype="msp"/>
+            <param name="references" value="similarity/rcx_gc-ei_ms_subset.msp" ftype="msp"/>
+            <param name="tolerance" value="60.0" />
+            <output name="scores_out" value="metadata_match/ri_match_60.json" ftype="json" />
+        </test>
+        <test>
+            <param name="use_scores" value="True"/>
+            <param name="scores_in" value="metadata_match/ri_match_60.json" ftype="json"/>
+            <param name="tolerance" value="20.0" />
+            <param name="join_type" value="inner" />
+            <output name="scores_out" value="metadata_match/inner_join_ri_20.json" ftype="json" />
+        </test>
+        <test>
+            <param name="use_scores" value="True"/>
+            <param name="scores_in" value="metadata_match/ri_match_60.json" ftype="json"/>
+            <param name="tolerance" value="20.0" />
+            <param name="join_type" value="left" />
+            <output name="scores_out" value="metadata_match/left_join_ri_20.json" ftype="json" />
+        </test>
+    </tests>
```

Comment:
Missing test case for `equal match` option.

Reply:
solved in https://github.com/RECETOX/galaxytools/pull/321/commits/5c4c6864d054d50cedc0443772c9ff4359434d92